File: 1561-maximum-number-of-coins-you-can-get/1561-maximum-number-of-coins-you-can-get.cpp
class Solution {
public:
    int maxCoins(vector<int>& piles) {
      int n=piles.size(),sum=0,counts=0;
        sort(piles.begin(),piles.end());
        //we do n/3 to leave 1/3 of the elements to bob and iterate the rest 2/3 with Alison
        for(int i=n/3; i<n; i+=2){
            sum+=piles[i];
        }
        return sum;
       
    }
};

File: 1561-maximum-number-of-coins-you-can-get/1561-maximum-number-of-coins-you-can-get.py
class Solution:
    def maxCoins(self, piles: List[int]) -> int:
        length = len(piles)
        sum = 0
        piles.sort()
        for index in range(length-2, (length//3)-1, -2):
            sum += piles[index]
            
        return sum